test/framework/src/main/java/org/elasticsearch/test/rest/yaml/ESClientYamlSuiteTestCase.java

```diff
@@ -407,13 +395,16 @@ protected boolean randomizeContentType() {
     }
 
     /**
-     * Sniff the cluster for host metadata.
+     * Sniff the cluster for host metadata and return a
+     * {@link RestClientBuilder} for a client with that metadata.
      */
-    private List<Node> sniffHostMetadata() throws IOException {
+    protected final RestClientBuilder sniffHostMetadata() throws IOException {
```

Comment:
I would rename this method to something like `getClientBuilderWithSniffedHosts`

Reply:
I'll do it!